### PR TITLE
Add DiscordBot README for better new dev onboarding process

### PR DESCRIPTION
**Pull request by rodonguyen**
Readme now:
image

### CODE REVIEW

**Comment by nqngo** [comment]
Hi @rodonguyen , please update the title and description to summarise what you changed. Title format should be:

```shell
$VERB $OBJECT $REASON
```

For example, Add Getting Started to README.md

File: README_DISCORDBOT.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Discord Bot Development
+
+## Getting started
+- Create a bot in discord https://interactions-py.github.io/interactions.py/Guides/02%20Creating%20Your%20Bot/
+- Create your own discord server, add that bot to a test server (feel free to use )
+- Implement bot logics (and add bot token to your code too)
+- Now, run the bot code locally and test the bot on your server
+
+## Auto formatter for Python
```
Comment (samhwang) [inline README_DISCORDBOT.md:9]: Please leave this for now, until we decided on a unified formatting tool between `ruff`, or use `black` and `flake8`. Possible a @nqngo question?

Comment (nqngo) [reply README_DISCORDBOT.md:9]: I'm partial to linting tech. However, I am leaning towards the tried and true `black` and `flake8`. Simply because you might be able to slap an upstream `devcontainer.json` and go. I'd imagine you have to do a `apt install` at least on a debian container for `ruff`.

Comment (samhwang) [reply README_DISCORDBOT.md:9]: nah `ruff` can be installed via poetry, just like `black` and `flake8`. It's quickly becoming the new standard now. 
image
https://github.com/astral-sh/ruff?tab=readme-ov-file#whos-using-ruff

Perhaps take this out into another discussion?

Comment (nqngo) [reply README_DISCORDBOT.md:9]: Sounds good. Tested it out in my devcontainer and works wonderfully. Will probably put it as a separated issue. The integration with VSCode is great too.

Comment (rodonguyen) [reply README_DISCORDBOT.md]: Hey I just put Ruff usage instructions in this PR. Is it ok?

File: README.md [hunk 2]
```diff
@@ -40,7 +40,8 @@ poetry add your_new_package_name
 We use [Ruff](https://github.com/astral-sh/ruff) for linting and code formating. Ruff is already included and installed when you use `poetry install`. Follow the [instructions here](https://github.com/astral-sh/ruff) to use Ruff.
 
 **TL;DR:**  
-Just use this (Lint all files in the current directory and any subdirectories) before committing new code.
+Just use this before committing new code.
 ```
-ruff check .
+ruff check .        # Lint all
+ruff format .       # Format all
```
Comment (samhwang) [inline README.md:46]:
```suggestion
poetry run ruff check .        # Lint all
poetry run ruff format .       # Format all
```

File: discord_bot/bot.py [hunk 3]
```diff
@@ -8,58 +8,58 @@
 
 @interactions.listen()
 async def on_ready():
-  print("Ready")
-  print(f"This bot is owned by {bot.owner}")
+    print("Ready")
```
Comment (samhwang) [inline discord_bot/bot.py:11]: Rebase your branch on to master and get the latest ruff rules. We have updated to use 2 spaces instead of 4. #21

File: README.md [hunk 4]
```diff
@@ -22,3 +22,26 @@ When you’re done working in the virtual environment, simply type:
...
+To add missing package to the package list:
+```
+poetry add your_new_package_name
+```
+
+## Discord Bot Development
+
+### Getting started
+- [Create a bot in discord](https://interactions-py.github.io/interactions.py/Guides/02%20Creating%20Your%20Bot/)
+- Create your own discord server, add that bot to a test server (feel free to use )
```
Comment (samhwang) [inline README.md:35]:
```suggestion
- Add that bot to a test server.
  - When making the invite through OAuth2 URL Generator, make sure to enable `bot` and `applications.commands` options.
```

File: README.md [hunk 5]
```diff
@@ -22,3 +22,26 @@ When you’re done working in the virtual environment, simply type:
...
+- Implement bot logics (and add bot token to your code too)
```
Comment (samhwang) [inline README.md:36]:
```suggestion
- Copy out `.env.example` into `.env`, and fill in the `DISCORD_BOT_TOKEN` and `DISCORD_GUILD_ID`
```

File: README.md [hunk 6]
```diff
@@ -22,3 +22,26 @@ When you’re done working in the virtual environment, simply type:
...
+- Now, run the bot code locally and test the bot on your server
+
+### Auto formatter for Python
+We use [Ruff](https://github.com/astral-sh/ruff) for linting and code formating. Ruff is already included and installed when you use `poetry install`. Follow the [instructions here](https://github.com/astral-sh/ruff) to use Ruff.
```
Comment (samhwang) [inline README.md:40]:
```suggestion
We use [Ruff](https://github.com/astral-sh/ruff) for linting and code formating and is installed via `poetry`. Follow the [instructions here](https://github.com/astral-sh/ruff) to use Ruff.
```

File: README.md [hunk 7]
```diff
@@ -22,3 +22,26 @@ When you’re done working in the virtual environment, simply type:
 ```
 exit
 ```
+
+To add missing package to the package list:
```
Comment (samhwang) [inline README.md:26]: I don't feel like we should add this. It's pretty much standard by now.

File: README.md [hunk 8]
```diff
@@ -22,3 +22,26 @@ When you’re done working in the virtual environment, simply type:
...
+```
+poetry add your_new_package_name
+```
+
```
Comment (samhwang) [inline README.md:30]: I don't feel like we should add this. It's pretty much standard by now.

```suggestion
```

File: README.md (hunk 6 above)
Comment (nqngo) [reply README.md:40]: Probably should just remove. Linting should go into CONTRIBUTING.md

File: pyproject.toml [hunk 9]
```diff
@@ -11,6 +11,7 @@ litellm = {extras = ["proxy"], version = "^1.23.7"}
 openai = "^1.11.1"
 python-dotenv = "^1.0.1"
 discord-py-interactions = "5.11.0"
+ruff = "^0.2.1"
```
Comment (nqngo) [inline pyproject.toml:14]: Ruff is already merged into dev dependencies. Please remove from commit.

File: poetry.lock
Comment (nqngo) [inline poetry.lock:1]: Unstage and remove from change. Poetry.lock should already been updated by other PR.

**Review by nqngo — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -22,3 +22,13 @@ When you’re done working in the virtual environment, simply type:
...
+## Discord Bot Development
+
+### Getting started
+- [Create a bot in discord](https://interactions-py.github.io/interactions.py/Guides/02%20Creating%20Your%20Bot/)
+- Add that bot to a test server.
+  - When making the invite through OAuth2 URL Generator, make sure to enable `bot` and `applications.commands` options.
+  - Follow the [Invite your bot](https://interactions-py.github.io/interactions.py/Guides/02%20Creating%20Your%20Bot/) section for reference.
+- Copy out `.env.example` into `.env`, and fill in the `DISCORD_BOT_TOKEN` and `DISCORD_GUILD_ID`
+- Now, run the bot code locally and test the bot on your server
```
Comment (samhwang) [inline README.md:34]: Add an extra line at the end

**Review by samhwang — CHANGES_REQUESTED** [review]

**Review by samhwang — APPROVED** [review]
